packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -174,47 +176,48 @@ const tablist = style({
 });
 
 export function TabList<T extends object>(props: TabListProps<T>) {
-  let {density, isDisabled, disabledKeys, orientation, labelBehavior, onFocus} = useContext(InternalTabsContext) ?? {};
-  let {showItems} = useContext(CollapseContext) ?? {};
+  let {showTabs} = useContext(CollapseContext) ?? {};
+  if (showTabs) {
+    return <TabListInner {...props} />;
+  }
+}
+
+function TabListInner<T extends object>(props: TabListProps<T>) {
+  let {density, isDisabled, disabledKeys, orientation, labelBehavior} = useContext(InternalTabsContext) ?? {};
   let state = useContext(TabListStateContext);
   let [selectedTab, setSelectedTab] = useState<HTMLElement | undefined>(undefined);
   let tablistRef = useRef<HTMLDivElement>(null);
 
   useLayoutEffect(() => {
-    if (tablistRef?.current && showItems) {
+    if (tablistRef?.current) {
       let tab: HTMLElement | null = tablistRef.current.querySelector('[role=tab][data-selected=true]');
 
       if (tab != null) {
         setSelectedTab(tab);
       }
-    } else if (tablistRef?.current) {
-      let picker: HTMLElement | null = tablistRef.current.querySelector('button');
-      if (picker != null) {
-        setSelectedTab(picker);
-      }
     }
-  }, [tablistRef, state?.selectedItem?.key, showItems]);
+  }, [tablistRef, state?.selectedItem?.key]);
 
-  let prevFocused = useRef<boolean | undefined>(false);
-  useLayoutEffect(() => {
-    if (!showItems && !prevFocused.current && state?.selectionManager.isFocused) {
-      onFocus();
-    }
-    prevFocused.current = state?.selectionManager.isFocused;
-  }, [state?.selectionManager.isFocused, state?.selectionManager.focusedKey, showItems]);
+  // let prevFocused = useRef<boolean | undefined>(false);
```

Comment:
Noticed that both in the previous implementation and this one, swapping between the components will lose focus. I swear I had that addressed at one point, but we should probably include something so that doesn't occur. I suspect it was linked with this section of code originally

Reply:
It was definitely this code, however, it didn't restore back when the picker expanded back to the tabs, so I've omitted it for now to simplify getting the PR in
We can add that behavior later